src/main/java/org/snomed/snowstorm/core/data/services/BranchReviewService.java

```diff
@@ -468,11 +471,141 @@ private void lookupBranchReviewConceptChanges(BranchReview branchReview) {
+		Set<Long> contradictoryConceptIdentifiers = new HashSet<>();
+
+		// Deleted on Source but modified on Target
+		String sourcePath = source.getPath();
+		String targetPath = target.getPath();
+		Set<String> versionsReplaced = branchService.findLatest(targetPath).getVersionsReplaced(Concept.class);
+		if (!versionsReplaced.isEmpty()) {
+			List<Long> concepts = new ArrayList<>(mapInternalIdentifiersToConceptIdentifiers(versionsReplaced, sourcePath));
+			removeIfFoundOnPath(concepts, targetPath); // If found, then not deleted
+
+			contradictoryConceptIdentifiers.addAll(concepts);
+		}
+
+		// Modified on Source but deleted on Target
+		List<String> conceptIdsWithModifiedDescriptions = new ArrayList<>();
+		try (final SearchHitsIterator<Description> stream = elasticsearchTemplate.searchForStream(
+				newSearchQuery(versionControlHelper.getUpdatesOnBranchDuringRangeCriteria(sourcePath, getStart(branchReview, source, target), source.getHead())).build(), Description.class)
+		) {
+			stream.forEachRemaining(hit -> conceptIdsWithModifiedDescriptions.add(hit.getContent().getConceptId()));
+		}
+
+		Map<Long, Date> conceptsEndedOnTarget = new HashMap<>();
+		for (String conceptId : conceptIdsWithModifiedDescriptions) {
+			try (final SearchHitsIterator<Concept> stream = elasticsearchTemplate.searchForStream(new NativeSearchQueryBuilder()
+					.withSort(SortBuilders.fieldSort(Description.Fields.START).order(SortOrder.DESC))
+					.withQuery(
+							boolQuery()
+									.must(matchQuery(Description.Fields.CONCEPT_ID, conceptId))
+									.must(matchQuery(Description.Fields.PATH, targetPath))
+					)
+					.build(), Concept.class)) {
+				stream.forEachRemaining(hit -> {
+					Concept concept = hit.getContent();
+
+					boolean endedOnTarget = concept.getEnd() != null && targetPath.equals(concept.getPath());
+					if (endedOnTarget) {
+						conceptsEndedOnTarget.put(parseLong(conceptId), concept.getEnd());
+					}
+				});
+			}
+		}
+
+		Iterator<Map.Entry<Long, Date>> iterator = conceptsEndedOnTarget.entrySet().iterator();
+		String targetParentPath = PathUtil.getParentPath(targetPath);
+		if (targetParentPath != null) {
+			// Confirm reason for ending documents, i.e. promotion or deletion
+			removeIfPromoted(conceptsEndedOnTarget, targetParentPath);
+		}
+
+		contradictoryConceptIdentifiers.addAll(conceptsEndedOnTarget.keySet());
+
+		return contradictoryConceptIdentifiers;
+	}
+
+	private Set<Long> mapInternalIdentifiersToConceptIdentifiers(Set<String> internalIds, String path) {
+		Set<Long> conceptIdentifiers = new HashSet<>();
+		for (String internalId : internalIds) {
+			try (final SearchHitsIterator<Concept> stream = elasticsearchTemplate.searchForStream(new NativeSearchQueryBuilder()
+					.withSort(SortBuilders.fieldSort(Description.Fields.START).order(SortOrder.DESC))
+					.withPageable(PageRequest.of(0, 1))
+					.withQuery(
+							boolQuery()
+									.must(matchQuery("_id", internalId))
```

Comment:
@dmcgihtsdo We should use termQuery instead of matchQuery as these fields are all keyword type. We only use matchQuery for full text search.

Reply:
Thanks @CoderMChu 

I have done this.